clang/lib/AST/TypePrinter.cpp

```diff
@@ -2062,6 +2066,7 @@ void TypePrinter::printBTFTagAttributedAfter(const BTFTagAttributedType *T,
 void TypePrinter::printHLSLAttributedResourceBefore(
     const HLSLAttributedResourceType *T, raw_ostream &OS) {
   printBefore(T->getWrappedType(), OS);
+  printAfter(T->getWrappedType(), OS);
```

Comment:
This looks suspicious - should `printHLSLAttributedResourceBefore` really be calling `printAfter`?

Reply:
Yea, we need to print the whole wrapped type before printing the attributes. I will move the attribute printing to printHLSLAttributedResourceAfter.